### PR TITLE
Add --private flag to edit_url.

### LINKED ISSUE #3185: Feature request: Add --private option for edit-url
Overall, it would be nice if `edit-url` had all the same options as `open` does.

### PR DESCRIPTION
**Pull request by rcorre**
The command :edit-url --private (or :edit-url -p) will spawn a new
private window with the url input from the editor.

I had to add 'Given I have a fresh instance' to the feature file to
ensure tests were not interfering.

Resolves #3185.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3200)

### CODE REVIEW

**Comment by The-Compiler** [comment]
Maybe `--related` would make sense too? Also, I'm wondering whether `edit_url` could just delegate to the `open` command, but I didn't look at the code yet.

**Comment by rcorre** [comment]
Not sure what you mean by `--related`. And `edit-url` does call `openurl` through a callback (`_open_if_changed`)

**Comment by The-Compiler** [comment]
`:open` has a `--related` switch which (when used with `--tab`) uses `tabs.new_position.related` instead of `tabs.new_position.unrelated` (i.e. it opens it close to the current tab).

File: qutebrowser/browser/commands.py [hunk 1]
```diff
@@ -2167,9 +2171,13 @@ def _open_if_changed(self, url=None, old_url=None, bg=False, tab=False,
             bg: Open in a new background tab.
             tab: Open in a new tab.
             window: Open in a new window.
+            private: Open a new window in private browsing mode.
+            related: If opening a new tab, position the tab as related to the
+                     current one (like clicking on a link).
         """
-        if bg or tab or window or url != old_url:
-            self.openurl(url=url, bg=bg, tab=tab, window=window)
+        if bg or tab or window or private or related or url != old_url:
```
Comment (The-Compiler) [inline qutebrowser/browser/commands.py]: pylint complains with `too-many-boolean-expressions` here, but I'm inclined to just turn that check off. What do you think?

Comment (rcorre) [reply qutebrowser/browser/commands.py]: agreed

**Comment by rcorre** [comment]
The [failures](https://travis-ci.org/qutebrowser/qutebrowser/jobs/293626647#L989) appear to be unrelated

**Comment by The-Compiler** [comment]
Thanks! They indeed are unrelated... I'd really like to move some of those end2end tests to smaller unit tests (probably after #1129 and #640.

**Pull request merged**